R/stroke.R

```diff
@@ -130,38 +130,40 @@ get_linked_nodes <- function(node_id, segment_id, segments) {
 
 #' @noRd
 best_link <- function(nodes, segments, links, angle_threshold = 0) {
+    
+    # convert nodes to a matrix for faster indexing
+    nodes <- as.matrix(nodes[c("x", "y")])
+    
+    best_links <- array(integer(), dim = dim(segments))
+    colnames(best_links) <- c("start", "end")
+    
+    angle_threshold_rad <- angle_threshold / 180 * pi  # convert to radians
```

Comment:
I am thinking whether it makes more sense to do the conversion to radians right at the beginning of the program in the main function. We can then pass the angle with  the right unit through the functions, where we can then refer to is simply as to `angle_threshold` (instead of `angle_threshold_rad`).